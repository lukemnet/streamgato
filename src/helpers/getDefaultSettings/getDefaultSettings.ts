interface FormField {
  [key:string]: string | number;
  value: string | number;
}

interface FormParam {
  [key:string]: FormField;
}

interface InitialValues {
  [key:string]: string | number;
}

const getDefaultSettings = (params: FormParam) => {
  const fieldNames = Object.keys(params);
  let defaultSettings = {} as InitialValues;

  fieldNames.forEach(fieldName => {
    const { shorthand, value }  = params[fieldName];
    defaultSettings[shorthand] = value;
  });

  return defaultSettings;
}

export default getDefaultSettings;
