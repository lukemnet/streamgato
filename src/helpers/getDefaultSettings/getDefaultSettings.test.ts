import getDefaultSettings from './getDefaultSettings';
import { WidgetParams } from 'types';

const testParams = {
  hours: {
    type: "number",
    label: "Hours",
    shorthand: "h",
    value: 0,
    min: 0,
  },
  minutes: {
    type: "number",
    label: "Minutes",
    shorthand: "m",
    value: 0,
    min: 0,
    max: 59,
    info: "0 to 59"
  },
  seconds: {
    type: "number",
    label: "Seconds",
    shorthand: "s",
    value: 0,
    min: 0,
    max: 59,
    info: "0 to 59",
  },
  background: {
    type: "string",
    label: "Background color",
    shorthand: "bg",
    value: "#000000",
    info: "Hex, RGB or RGBA. 'transparent' for no color"
  },
  color: {
    type: "string",
    label: "Text color",
    shorthand: "col",
    value: "#ffffff",
    info: "Hex, RGB or RGBA"
  }
} as WidgetParams;

it('should create default settings object', () => {
  expect(getDefaultSettings(testParams)).toMatchSnapshot();
})