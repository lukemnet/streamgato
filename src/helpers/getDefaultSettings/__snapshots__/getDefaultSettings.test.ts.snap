// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create default settings object 1`] = `
Object {
  "bg": "#000000",
  "col": "#ffffff",
  "h": 0,
  "m": 0,
  "s": 0,
}
`;
